packages/msal-interactor/src/MsalInteractor.ts

```diff
@@ -2,105 +2,230 @@
-		private scopes: string[],
-		private authDisabled = false,
-	) {}
+	/**
+	 * Singleton. Use MsalInteractor.configure(options)
+	 * and MsalInteractor.getInstance()
+	 */
+	private constructor({ msalConfig, oidcScopes }: MsalInteractorOptions) {
+		this._msalConfig = msalConfig
+		this._oidcScopes = oidcScopes
+		this._msalInstance = new PublicClientApplication(this._msalConfig)
+
+		// On page load/instantiation, check if responding to redirect auth flow
+		const pageLoadHandler = this._msalInstance
+			.handleRedirectPromise()
+			.then(this._handleAuthResponse)
+
+		this.isAuthenticated = () => pageLoadHandler.then(res => !!res)
+	}
+
+	public static configure = (options: MsalInteractorOptions): void => {
+		MsalInteractor._options = options
+	}
+
+	public static getInstance = (): MsalInteractor => {
+		if (MsalInteractor._instance) {
+			return MsalInteractor._instance
+		}
+
+		if (!MsalInteractor._options) {
+			throw new TypeError(
+				`Cannot instantiate MsalInteractor without configuration options. Call configure() before getInstance()`,
+			)
+		}
+
+		MsalInteractor._instance = new MsalInteractor(MsalInteractor._options)
+		return MsalInteractor._instance
+	}
 
 	/**
-	 * Gets the MSAL Instance
+	 * User login.
+	 *
+	 * Attempt to login silently with current valid session
+	 * before logging in using either popUp or redirect flow.
 	 */
-	public get instance(): UserAgentApplication {
-		if (this._instance != null) {
-			return this._instance
+	public login = async (
+		options: MsalInteractorLoginOptions = {},
+	): Promise<AuthenticationResult | null> => {
+		const { usePopup = false } = options
+
+		// try to login silently
+		if (await this.isAuthenticated()) {
+			const authResult = await this._getAuthResult()
+			if (authResult) {
+				return authResult
+			}
 		}
 
-		const result = new UserAgentApplication(this.msalConfig)
-		if (result != null) {
-			result.handleRedirectCallback((err, res) => {
-				console.log('MSAL handling redirect callback', err, res)
-			})
+		if (usePopup) {
+			const authResult = this._msalInstance
+				.loginPopup({
+					scopes: this._oidcScopes,
+				})
+				.then(this._handleAuthResponse)
+
+			this.isAuthenticated = () => authResult.then(res => !!res)
+			return authResult
+		} else {
+			return this._msalInstance
+				.loginRedirect({ scopes: this._oidcScopes })
+				.then(r => null)
+		}
+	}
+
+	/**
+	 * User logout
+	 */
+	public logout = async (): Promise<void> => {
+		this.isAuthenticated = async () => false
+		if (!this._accountInfo) {
+			return
 		}
-		this._instance = result
-		return result
+		const logoutRequest = {
+			account: this._accountInfo,
+		}
+
+		return this._msalInstance.logout(logoutRequest)
 	}
 
 	/**
-	 * Performs a user login
+	 * Get user ID claims.
 	 */
-	public async login(): Promise<AuthResponse | undefined> {
-		if (this.authDisabled || this.suppressTokenRequest()) {
-			return Promise.resolve(undefined)
+	public getIdTokenClaims = async (): Promise<
```

Comment:
Do we not need to check if we're in an iframe context anymore? If so, that's pretty nice.